reroute/src/Web/Routing/Combinators.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+module Web.Routing.Combinators where
+
+import Data.HVect (Append)
+import Data.String
+import Data.Typeable (Typeable)
+import Web.PathPieces
+import qualified Data.Text as T
+
+import Web.Routing.SafeRouting
+
+data PathState = Open | Closed
+
+data SafePath (as :: [*]) (pathState :: PathState) where
```

Comment:
I think it would be better to keep the user facing Path type named `Path`, and rename the old (now internal) path to something else like `PathInternal`